libcontainer/specconv/spec_linux.go

```diff
@@ -366,6 +366,39 @@ func CreateLibcontainerConfig(opts *CreateOpts) (*configs.Config, error) {
 			}
 		}
 	}
+
+	// Set the host UID that should own the container's cgroup.
+	// This must be performed after setupUserNamespace, so that
+	// config.HostRootUID() returns the correct result.
+	//
+	// Only set it if the container will have its own cgroup
+	// namespace and the cgroupfs will be mounted read/write.
+	//
+	hasCgroupNS := config.Namespaces.Contains(configs.NEWCGROUP) && config.Namespaces.PathOf(configs.NEWCGROUP) == ""
+	hasRwCgroupfs := false
+	for _, m := range config.Mounts {
+		if m.Source == "cgroup" && m.Destination == "/sys/fs/cgroup" && (m.Flags&unix.MS_RDONLY) == 0 {
+			hasRwCgroupfs = true
+			break
+		}
+	}
```

Comment:
In case `hasCgroupNS` is false, the results of this check is useless. This path can be quite common (for cgroup v1) so it probably makes sense to optimize it.

Reply:
OK, I'll guard this loop behind `if hasCgroupNS`.